### ISSUE TITLE
Default? values for primitives in classes ignored in 0.9.8

### ISSUE THREAD

**Issue by cacsar** [leangen/graphql-spqr#192]
Similar to #163 we hit something where all primitives in fields were being marked as NonNull even though they had default values set in their definition. Since we upgraded to 0.9.8 from 0.9.3 along with the data classes changing from lombok to Kotlin I'm not certain where the problem lies (probably with us). I'm not yet certain of our motivation behind the update.

By following your advice in #163 the I saw relevant fields change in the GraphQL Schema from Boolean! to Boolean and Int! to Int and fixed the problem we encountered in the immediate term.

That is I registered

```
public class NonNullMapperNoPrimitiveFields extends NonNullMapper
{
    @Override
    public boolean supports(AnnotatedType type) {
        //Means we don't support any primitive types.
        if(ClassUtils.getRawType(type.getType()).isPrimitive())
        {
            //This way of writing so it's easier to put in break points.
            return false;
        }
        return super.supports(type);
    }
}
```
and enabled it with `.withTypeMappers((conf, current) -> current.replace(NonNullMapper.class, new NonNullMapperNoPrimitiveFields()))`

**Comment by cacsar** [comment]
That is we had a class used nested in the input of a GraphQL mutation contains a field 'hidden' previously this could be omitted from the GraphQL input, after the update it became necessary to send `hidden : false` until the workaround above was used. In the kotlin class we have `var isHidden : Boolean = false`

**Comment by kaqqao** [comment]
Sorry for not answering this sooner 😖
This is a bug in 0.9.8 that has been fixed in the just-released 0.9.9.
Primitives are intentionally mapped as non-null as their nature does not allow them to be nullable, but making them non-null despite a default value being present was a bug...
